### PR TITLE
Add Dash live updates example (Live Bitcoin Tracker)

### PR DESCRIPTION
**Pull request by ellieko**
Added a Real-time Live Bitcoin Tracker demo in examples/dash/dash-live-updates as an example of displaying real-time data in Dash

In the examples/dash/dash-live-updates directory:

For README:
- README.md
- dashboard.png

For REST API method (not used in the blog post or demo)
- rest_api_dash.py (REST API method)

For WebSocket API method  (used in the blog post and demo)
- app.py (Dash app for WebSocket method)
- websocket_backend.py (Data Fetching and Storage for WebSocket method)
- requirements.txt (libraries needed for WebSocket method)

The detailed explanation is in the draft blog post. The dash app is deployed but stopped for now, because testing the demo locally would be only possible with a remote DB requirement. I can share my DB connection details via Slack if that's okay.

----
📚 Documentation preview 📚: https://ploomber-doc--269.org.readthedocs.build/en/269/

### CODE REVIEW

File: examples/dash/dash-live-updates/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+```
+DB_USER=<USERNAME>
+DB_PASSWORD=<PASSWORD>
+DB_NAME=<DATABASE_NAME>
+DB_HOST=<HOST>
+DB_PORT=<PORT>
+```
+
+### 3. Backend Setup
+`websocket_backend.py`: connects to the Binance WebSocket API to receive real-time trade data for Bitcoin (BTC/USDT), and stores it in a PostgreSQL database equipped with the TimescaleDB extension.
```
Comment (neelasha23) [inline examples/dash/dash-live-updates/README.md:26]: Mention the architecture briefly. that the local script stores the data in cloud and the monitoring app will connect to this DB for fetching the updates.

File: examples/dash/dash-live-updates/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+For local testing, you need to modify `websocket_backend.py` according to your database type. By default, it uses TimescaleDB (a PostgreSQL extension).
+
+If you prefer to use SQLite, you can modify the script to use `asyncio`, `aiosqlite`, and `sqlite3` libraries, along with SQL statements. If needed, an alternative script with SQLite can be added.
+
+Run the backend script:
+```sh
+python websocket_backend.py
+```
+
+### 4. Frontend Setup
```
Comment (neelasha23) [inline examples/dash/dash-live-updates/README.md:37]: Add a Ploomber Cloud deployment section where you can add details like the files required for zip (all except local script and README), add link to Dash deployment guide. mention about secrets and add reference.

File: examples/dash/dash-live-updates/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+### 2. Save Database Connection Details as Environment Variables
+```
+DB_USER=<USERNAME>
+DB_PASSWORD=<PASSWORD>
+DB_NAME=<DATABASE_NAME>
+DB_HOST=<HOST>
+DB_PORT=<PORT>
+```
+
+### 3. Backend Setup
```
Comment (neelasha23) [inline examples/dash/dash-live-updates/README.md:25]: Add steps for setting up TimescaleDB with screenshots.

File: examples/dash/dash-live-updates/README.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+### 1. Development Environment Setup
+```sh
+conda create --name YOUR_ENV_NAME python=3.11
+conda activate YOUR_ENV_NAME
+pip install -r requirements.txt
+```
+
+### 2. Save Database Connection Details as Environment Variables
+```
+DB_USER=<USERNAME>
```
[the remark this answers is not in the document]
Comment (ellieko) [reply examples/dash/dash-live-updates/README.md:41]: [fixed](https://github.com/ploomber/doc/pull/269/files#diff-cd1a6fcc32f127145ab61284c47ea54083a31b43a7ef552054afbdee033a3d83R38)

File: examples/dash/dash-live-updates/README.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+DB_HOST=<HOST>
+DB_PORT=<PORT>
+```
+
+### 3. Backend Setup
+`websocket_backend.py`: connects to the Binance WebSocket API to receive real-time trade data for Bitcoin (BTC/USDT), and stores it in a PostgreSQL database equipped with the TimescaleDB extension.
+
+For local testing, you need to modify `websocket_backend.py` according to your database type. By default, it uses TimescaleDB (a PostgreSQL extension).
+
+If you prefer to use SQLite, you can modify the script to use `asyncio`, `aiosqlite`, and `sqlite3` libraries, along with SQL statements. If needed, an alternative script with SQLite can be added.
```
[the remark this answers is not in the document]
Comment (ellieko) [reply examples/dash/dash-live-updates/README.md]: [fixed](https://github.com/ploomber/doc/pull/269/files#diff-cd1a6fcc32f127145ab61284c47ea54083a31b43a7ef552054afbdee033a3d83R25)

File: examples/dash/dash-live-updates/README.md (hunk 1 above)
Comment (ellieko) [reply examples/dash/dash-live-updates/README.md]: [fixed](https://github.com/ploomber/doc/pull/269/files#diff-cd1a6fcc32f127145ab61284c47ea54083a31b43a7ef552054afbdee033a3d83R7-R12)

File: examples/dash/dash-live-updates/README.md (hunk 2 above)
Comment (ellieko) [reply examples/dash/dash-live-updates/README.md]: [fixed](https://github.com/ploomber/doc/pull/269/files#diff-cd1a6fcc32f127145ab61284c47ea54083a31b43a7ef552054afbdee033a3d83R67-R124)

File: examples/dash/dash-live-updates/README.md (hunk 3 above)
Comment (ellieko) [reply examples/dash/dash-live-updates/README.md]: [fixed](https://github.com/ploomber/doc/pull/269/files#diff-cd1a6fcc32f127145ab61284c47ea54083a31b43a7ef552054afbdee033a3d83R27-R49)

**Comment by ellieko** [comment]
Hi @neelasha23, @bryannho, this is also ready for review. Thanks a lot! :))

**Review by bryannho — COMMENTED** [review]
@ellieko I saw that `rest_api_dash.py` is included but isn't used. I think it could be useful to keep it in case users wanted to try that method. Can you add a note in the `README.md` explaining what the file is and that it isn't used?

**Comment by ellieko** [comment]
> @ellieko I saw that `rest_api_dash.py` is included but isn't used. I think it could be useful to keep it in case users wanted to try that method. Can you add a note in the `README.md` explaining what the file is and that it isn't used? otherwise, lgtm

Good point, thanks! [Added here](https://github.com/ploomber/doc/pull/269/files#diff-cd1a6fcc32f127145ab61284c47ea54083a31b43a7ef552054afbdee033a3d83R14)

**Review by neelasha23 — APPROVED** [review]

**Pull request merged**